app/services/store_sp_metadata_in_session.rb

```diff
@@ -5,6 +5,8 @@ def initialize(session:, request_id:)
   end
 
   def call
+    return unless sp_request
```

Comment:
This PR gives me a little heartburn.  If this is a genuine problem, we should at least *log* something about it so we know when it happens (and how often), and possibly enough information that we can identify what user is in this state.  Returning this error to the user with a 500 might be reasonable until we figure out what's causing it.  If the user is providing a broken/invalid request ID, maybe the right thing is just to generate another error with a 200 or 4xx response code.  The only thing I'd prefer we avoid is an uncaught exception that then pages us through New Relic.